Client dev server exits (502) when a lint error is present at startup

## Problem

A single ESLint **error** crashes the client dev server entirely — the container exits and nginx returns **502**. This is not the recoverable HMR error overlay; the Node process dies and stops serving.

Root cause is `vite-plugin-eslint2` with `lintOnStart: true` (`client/quasar.config.ts`). On startup it lints the whole project in the rollup `buildStart` hook and calls `PluginContext.error()` on any error, which throws and aborts Vite startup (`exit code 1`).

Combined with `eslint: { errors: true }` and error-level rules (e.g. the new `@intlify/vue-i18n/no-raw-text` from #2302), any committed lint error makes the dev server un-startable rather than just flagging it.

### Example failure
```
/app/client/src/components/tables/common/NameAvatarCell.vue
  19:8  error  raw text '—' is used  @intlify/vue-i18n/no-raw-text
✖ 1 problem (1 error, 0 warnings)
    at PluginContext.error ... vite-plugin-eslint2
error Command failed with exit code 1.
```

## Impact

- A lint error anywhere in the tree blocks **all** local development with an opaque 502.
- Failure mode is confusing: looks like an infra/proxy problem, not a lint problem.

## Proposed options

1. Decouple boot lint from server start — drop `lintOnStart: true` (lint still runs per-file via overlay + in CI).
2. Keep `lintOnStart` but make it non-fatal — configure the plugin so startup errors report to the overlay instead of throwing (e.g. `emitErrorAsWarning` / `failOnError: false` semantics), so the server stays up.
3. Rely on CI lint as the hard gate; keep the dev server resilient.

Lint should still **fail CI** — this is only about not killing the local dev server.

## Notes
- Boot-time lint has always been fatal-on-error; it simply never triggered until #2302 added an error-level rule that matched committed code.

--- a/client/quasar.config.ts
+++ b/client/quasar.config.ts
@@ -151,6 +151,12 @@ export default defineConfig(function (/* ctx */) {
           {
             lintOnStart: true,
             fix: false,
+            // Emit lint errors as warnings during local dev so a committed
+            // lint error surfaces in the overlay/console without aborting
+            // Vite startup (PluginContext.error() throws and kills the dev
+            // server -> 502). Lint is still a hard gate in CI via `yarn lint`,
+            // which runs the ESLint CLI independently of this plugin. See #2303.
+            emitErrorAsWarning: true,
           }
         ],
         [

